### DISCUSSION TITLE
Remove automatically added email body text on email attachment

### DISCUSSION THREAD

**Discussion by mlevy823** [sumatrapdfreader/sumatrapdf#4533] [Ideas]
It would be nice to have an option of removing the automatically generated text that gets put into the email body when attaching a document to an email. 

> 
> Your message is ready to be sent with the following file or link attachments:
> 
> Example.pdf
> 
> 
> Note: To protect against computer viruses, e-mail programs may prevent sending or receiving certain types of file attachments.  Check your e-mail security settings to determine how attachments are handled.

**Comment by kjk** [comment]
Sumatra isn't doing that. It's your e-mail program and it's impossible for Sumatra to change that.

**Reply by mlevy823** [reply]
Ah I see. It didn't happen with adobe acrobat so I just assumed it was sumatra. It seems that Windows is doing it (I am using Outlook). Apparently it is the "Send to" feature built into windows explorer using sendmail.dll.

**Comment by mlevy823** [comment]
I can confirm that this also happens in Thunderbird when attaching a pdf to an email via SumatraPDF, not just Outlook. I am using the prerelease with the CmdSendByEmail as a toolbar shortcut.

**Reply by wvutsr** [reply]
> I can confirm that this also happens in Thunderbird when attaching a pdf to an email via SumatraPDF, not just Outlook. I am using the prerelease with the CmdSendByEmail as a toolbar shortcut.

Have you got any working solution since?

**Reply by mlevy823** [reply]
I have not. I am not a programmer so it is out of my depth. I believe sumatrapdf currently uses SendMail.dll that is built into windows to attach the pdf. The SendMail.dll is the one putting in the text. The part of the code is in my comment below this one.

**Comment by mlevy823** [comment]
ExternalViewers.cpp, Lines 475-500:
```
// TODO: maybe use
// https://stackoverflow.com/questions/47639267/win32-c-sending-email-in-windows-10-by-invoking-default-mail-client
bool SendAsEmailAttachment(WindowTab* tab, HWND hwndParent) {
    if (!tab || !CanSendAsEmailAttachment(tab)) {
        return false;
    }


    // We use the SendTo drop target provided by SendMail.dll, which should ship with all
    // commonly used Windows versions, instead of MAPISendMail, which doesn't support
    // Unicode paths and might not be set up on systems not having Microsoft Outlook installed.
    ScopedComPtr<IDataObject> pDataObject(GetDataObjectForFile(tab->filePath, hwndParent));
    if (!pDataObject) {
        return false;
    }


    ScopedComPtr<IDropTarget> pDropTarget;
    if (!pDropTarget.Create(CLSID_SendMail)) {
        return false;
    }


    POINTL pt = {0, 0};
    DWORD dwEffect = 0;
    pDropTarget->DragEnter(pDataObject, MK_LBUTTON, pt, &dwEffect);
    HRESULT hr = pDropTarget->Drop(pDataObject, MK_LBUTTON, pt, &dwEffect);
    return SUCCEEDED(hr);
}
```

**Comment by wvutsr** [comment]
Is there any way to circumvent this? Or is this something that can never be fixed?

**Reply by GitHubRulesOK** [reply]
@wvutsr
From discussion above it is a default Part of Outlook via Windows dlls
so you need to use Microsoft VS to add your own alteration to Windows standard methods

**Reply by wvutsr** [reply]
Thanks for the quick reply. I will check if i can find a workaround then myself. Maybe a clear .textfield or something of that kind, after it has attached the file to a new email.
